concurrent/actor/Actor.java

```diff
@@ -89,20 +109,28 @@ public CompletableFuture<Void> complete(Consumer<ACTOR> consumer) {
 
         public <ANSWER> CompletableFuture<ANSWER> compute(Function<ACTOR, ANSWER> function) {
             assert actor != null : ERROR_ACTOR_NOT_SETUP;
-            CompletableFuture<ANSWER> future = new CompletableFuture<>();
-            executor.submit(
-                    () -> future.complete(function.apply(actor)),
-                    e -> {
-                        actor.exception(e);
-                        future.completeExceptionally(e);
-                    }
-            );
-            return future;
+            if (!actor.isTerminated) {
+                CompletableFuture<ANSWER> future = new CompletableFuture<>();
+                executor.submit(
+                        () -> {
+                            if (!actor.isTerminated) future.complete(function.apply(actor));
+                        },
+                        e -> {
+                            actor.exception(e);
+                            future.completeExceptionally(e);
+                        }
+                );
+                return future;
+            } else return null;
```

Comment:
we could debate whether this should return `null` or a `CompletedFuture` - what are the semantics of submitted a job to be run on a terminated Actor?